hue/hue.sh

```diff
@@ -24,127 +29,142 @@ function update_apt_get() {
 
   sed -i '/<\/configuration>/e cat oozie-site-patch.xml' \
-      /etc/oozie/conf/oozie-site.xml
+  /etc/oozie/conf/oozie-site.xml
 
   rm oozie-site-patch.xml
-  systemctl restart oozie
+  systemctl restart oozie || err "Unable to restart oozie"
+else
+  echo "oozie not installed, skipped configuring hue as an user proxy"
 fi
 
 # Fix webhdfs_url
-OLD_HDFS_URL='## webhdfs_url\=http:\/\/localhost:50070'
-NEW_HDFS_URL='webhdfs_url\=http:\/\/'"$(hdfs getconf -confKey  dfs.namenode.http-address)"
-sed -i 's/'"$OLD_HDFS_URL"'/'"$NEW_HDFS_URL"'/' /etc/hue/conf/hue.ini
+sed -i 's/'"${old_hdfs_url}"'/'"${new_hdfs_url}"'/' /etc/hue/conf/hue.ini
 
 # Uncomment every line containing localhost, replacing localhost with the FQDN
 sed -i "s/#*\([^#]*=.*\)localhost/\1$(hostname --fqdn)/" /etc/hue/conf/hue.ini
 
 # Comment out any duplicate resourcemanager_api_url fields after the first one
 sed -i '0,/resourcemanager_api_url/! s/resourcemanager_api_url/## resourcemanager_api_url/' \
-    /etc/hue/conf/hue.ini
+  /etc/hue/conf/hue.ini
 
 # Clean up temporary fles
 rm -rf hdfs-site-patch.xml core-site-patch.xml hue-patch.ini
 
 # Make hive warehouse directory
-hdfs dfs -mkdir /user/hive/warehouse
+if [[ ! -d /user/hive/warehouse ]]; then
+  hdfs dfs -mkdir /user/hive/warehouse || err "Unable to create folder"
+fi
 
 # Configure Desktop Database to use mysql
-OLD_SETTINGS='## engine=sqlite3(\s+)## host=(\s+)## port=(\s+)## user=(\s+)## password='
-NEW_SETTINGS='engine=mysql$1host=127.0.0.1$2port=3306$3user=hue$4password=hue-password'
-perl -i -0777 -pe 's/'"$OLD_SETTINGS"'/'"$NEW_SETTINGS"'/' /etc/hue/conf/hue.ini
+sed -i 's/'"${old_mysql_settings}"'/'"${new_mysql_settings}"'/' /etc/hue/conf/hue.ini
 
 # Comment out sqlite3 configuration
 sed -i 's/engine=sqlite3/## engine=sqlite3/' /etc/hue/conf/hue.ini
 
 # Set database name to hue
 sed -i 's/name=\/var\/lib\/hue\/desktop.db/name=hue/' /etc/hue/conf/hue.ini
 
-# Export passwords
-export MYSQL_PASSWORD='root-password'
-export HUE_PASSWORD='hue-password'
-
 # Create database, give hue user permissions
 mysql -u root -proot-password -e " \
   CREATE DATABASE hue; \
-  CREATE USER 'hue'@'localhost' IDENTIFIED BY '$HUE_PASSWORD'; \
-  GRANT ALL PRIVILEGES ON hue.* TO 'hue'@'localhost';"
+  CREATE USER 'hue'@'localhost' IDENTIFIED BY '${hue_password}'; \
+  GRANT ALL PRIVILEGES ON hue.* TO 'hue'@'localhost';" \
+  || err "Unable to create database"
 
 # Restart mysql
-systemctl restart mysql
+systemctl restart mysql || err "Unable to restart mysql"
 
 # Hue creates all needed tables
-/usr/lib/hue/build/env/bin/hue syncdb --noinput
-/usr/lib/hue/build/env/bin/hue migrate
+/usr/lib/hue/build/env/bin/hue syncdb --noinput || err "Database sync failed"
+/usr/lib/hue/build/env/bin/hue migrate || err "Data migration failed"
 
 # Restart servers
-systemctl restart hadoop-hdfs-namenode hadoop-yarn-resourcemanager hue mysql
+systemctl restart hadoop-hdfs-namenode || err "Unable to restart hadoop-hdfs-namenode"
```

Comment:
If systemctl already gives a good enough error message when it fails to start things, then I would leave it the way it was, since doing them together is presumably faster. (But feel free to push back on this)

Reply:
I've split this because of errors I got on this step. Since it's split I do not have any issues with this step.